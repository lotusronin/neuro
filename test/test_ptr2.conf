conf = {
'src' : 'test_ptr2.nro',
'expected' : 'test_ptr2.expected'
}
