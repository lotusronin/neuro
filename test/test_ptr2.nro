import stdio;

struct foo {
    num : u32;
}

fn main() : u32 {
    bar : foo;
    bar.num = 0;

    stdio::puts("bar.num is");
    stdio::printi(bar.num);
    stdio::putchar('\n');
    add1(&bar);
    stdio::puts("bar.num is now");
    stdio::printi(bar.num);
    stdio::putchar('\n');
    return 0;
}

fn add1(x : *foo) : void {
    (@x).num = (@x).num + 1;
    return;
}
