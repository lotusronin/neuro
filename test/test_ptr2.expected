bar.num is
0
bar.num is now
1
